### PR TITLE
CoreFxTesting code cleanup and remove performance support

### PR DESCRIPTION
**Pull request by ViktorHofer**
Porting some changes over from https://github.com/dotnet/arcade/pull/2037

- Remove performance test project support
- Error soft when test assembly not found
- Rename corefxtesting msbuild assembly property to something useful
- Paths with spaces support
- Code cleanup

### CODE REVIEW

File: src/Microsoft.DotNet.CoreFxTesting/build/TestProject.props [hunk 1]
```diff
@@ -2,16 +2,6 @@
 <!-- Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE file in the project root for more information. -->
 <Project>
 
-  <PropertyGroup Condition="'$(IsPerformanceTestProject)' == ''">
-    <IsPerformanceTestProject>false</IsPerformanceTestProject>
-    <IsPerformanceTestProject Condition="$(MSBuildProjectName.EndsWith('.PerformanceTests'))">true</IsPerformanceTestProject>
-  </PropertyGroup>
-
-  <PropertyGroup Condition="'$(IsIntegrationTestProject)' == ''">
```
Comment (adamsitnik) [inline src/Microsoft.DotNet.CoreFxTesting/build/TestProject.props]: @ViktorHofer I understand the removal of `IsPerformanceTestProject` but are you sure that `IsIntegrationTestProject` is not used anywhere?

File: src/Microsoft.DotNet.CoreFxTesting/build/TestProject.props [hunk 2]
```diff
@@ -25,9 +15,6 @@
     <IsTestSupportProject Condition="($(MSBuildProjectFullPath.Contains('\tests\')) OR $(MSBuildProjectFullPath.Contains('/tests/'))) AND '$(IsTestProject)' != 'true'">true</IsTestSupportProject>
     <IsTestProject Condition="'$(IsTestSupportProject)' == 'true'">true</IsTestProject>
 
-    <!-- Disable building Integration Test projects in LUT. -->
-    <BuildForLiveUnitTesting Condition="'$(BuildForLiveUnitTesting)' == '' and '$(IsIntegrationTestProject)' == 'true'">false</BuildForLiveUnitTesting>
```
Comment (adamsitnik) [inline src/Microsoft.DotNet.CoreFxTesting/build/TestProject.props]: similar: not related to performance, but removed on purpose?

File: src/Microsoft.DotNet.CoreFxTesting/build/TestProject.props (hunk 1 above)
Comment (ViktorHofer) [reply src/Microsoft.DotNet.CoreFxTesting/build/TestProject.props]: Yeah we removed the integration test functionality in the corefx script as it was never used.

File: src/Microsoft.DotNet.CoreFxTesting/build/TestProject.props (hunk 2 above)
Comment (ViktorHofer) [reply src/Microsoft.DotNet.CoreFxTesting/build/TestProject.props]: Yeah as IntegrationTest isn't used.

**Comment by ViktorHofer** [comment]
Flowing dependencies is problematic between arcade and corefx. In this case I want to test the changes with a nuget package on my own feed therefore I put all the changes together but grouped them in commits so that they are easier to review.

**Pull request merged**